### ISSUE TITLE
[Bug]: codebuild deploy provider ignores image property

### ISSUE THREAD

**Issue by AndreasAugustin** [awslabs/aws-deployment-framework#601]
### Is there an existing issue for this?

- [X] I have searched the existing issues
- #382 is maybe similar but is about using docker images

### Describe the bug

related to docs The deploy [provider codebuild](https://github.com/awslabs/aws-deployment-framework/blob/v3.1.2/docs/providers-guide.md#codedeploy) should have the same properties like using it within the build section.
there is a property `image` where you should be able to set the wanted image.
This is working nice using codebuild as build provider. But it seems that it does not work when you use it as deploy provider.

### Expected Behavior

the Codebuild image is set to the image configured within the properties section

### Current Behavior

It is just using the default image

### Steps To Reproduce

add a new build pipeline within the pipeline repository

add a deploy step like 

```yaml
      deploy:
        provider: codebuild
        properties:
          image: "STANDARD_5_0"
```

After the update has been done, check the image within codebuild

### Possible Solution

reuse the code like done for setting up the build part

### Additional Information/Context

Am not sure (yet) if this is already solved with the current version v3.2.0

**Update:** the bug still exists in V3.2.0

### ADF Version

v3.2.1

**Comment by sbkok** [comment]
I believe this is fixed by #628. In the upcoming v4.0 release I cannot reproduce this issue.

**Comment by sbkok** [comment]
Thank you for your patience. I am pleased to inform you that this issue has been [resolved in our latest release v4.0.0](https://github.com/awslabs/aws-deployment-framework/releases/tag/v4.0.0) just now.
I'm hereby closing this issue. Please open a new issue if you are experiencing any issues with the latest release.
